addons/point_of_sale/static/src/js/screens.js

```diff
@@ -2192,8 +2238,18 @@ var PaymentScreenWidget = ScreenWidget.extend({
                 self.gui.show_screen('receipt');
             });
         } else {
-            this.pos.push_order(order);
-            this.gui.show_screen('receipt');
+            var ordered = this.pos.push_order(order);
```

Comment:
you only managed the case push order, you also need to manage when there is an invoice, by modifying your server function with:  action_receipt_to_customer(self, name, email, receipt, invoice_id=False), and give it an invoice id in this specific case and add the invoice pdf to the email containing receipt.